### PR TITLE
Use DataObject and ProductListInterface constants

### PR DESCRIPTION
**Pull request by NiklasBr**
## Changes in this pull request  
Resolves no specific issue, only trying to avoid hard-coded strings where possible and the class constants are available.

### CODE REVIEW

File: bundles/AdminBundle/Controller/Admin/DataObject/DataObjectHelperController.php [hunk 1]
```diff
@@ -1256,7 +1256,7 @@ public function doExportAction(Request $request, LocaleServiceInterface $localeS
             $quotedIds[] = $list->quote($id);
         }
 
-        $list->setObjectTypes(['object', 'folder', 'variant']);
+        $list->setObjectTypes([DataObject::OBJECT_TYPE_OBJECT, DataObject::OBJECT_TYPE_FOLDER, DataObject::OBJECT_TYPE_VARIANT]);
```
Comment (dvesh3) [inline bundles/AdminBundle/Controller/Admin/DataObject/DataObjectHelperController.php:1259]: maybe we can use `DataObject::$types` at places where all types are defined?

File: bundles/EcommerceFrameworkBundle/IndexService/Worker/DefaultMysql.php [hunk 2]
```diff
@@ -121,7 +121,7 @@ public function updateIndex(IndexableInterface $object)
 
                 $virtualProductId = $subObjectId;
                 $virtualProductActive = $object->isActive();
-                if ($object->getOSIndexType() == 'variant') {
+                if ($object->getOSIndexType() == DataObject::OBJECT_TYPE_VARIANT) {
```
Comment (dvesh3) [inline bundles/EcommerceFrameworkBundle/IndexService/Worker/DefaultMysql.php:124]: `ProductListInterface::PRODUCT_TYPE_VARIANT` ?

File: bundles/EcommerceFrameworkBundle/IndexService/Worker/ProductCentricBatchProcessingWorker.php [hunk 3]
```diff
@@ -203,7 +203,7 @@ protected function getDefaultDataForIndex(IndexableInterface $object, $subObject
 
         $virtualProductId = $subObjectId;
         $virtualProductActive = $object->isActive();
-        if ($object->getOSIndexType() == 'variant') {
+        if ($object->getOSIndexType() == DataObject::OBJECT_TYPE_VARIANT) {
```
Comment (dvesh3) [inline bundles/EcommerceFrameworkBundle/IndexService/Worker/ProductCentricBatchProcessingWorker.php:206]: `ProductListInterface::PRODUCT_TYPE_VARIANT` ?

Comment (NiklasBr) [reply bundles/EcommerceFrameworkBundle/IndexService/Worker/ProductCentricBatchProcessingWorker.php:206]: I was unsure, but you are probably correct. Fixing.

**Review by dvesh3 — APPROVED** [review]

**Comment by dvesh3** [comment]
@NiklasBr you have some unrelated commits in your PR. could you please revert them?

**Comment by NiklasBr** [comment]
I'll see what I can do, rebaseing using the Github web UI was not as intuitive as I hoped.

**Comment by NiklasBr** [comment]
Replaced by #8781

**Pull request closed without merging**